wallet/how-to/use-non-evm-networks/starknet-snap/index.md

```diff
@@ -0,0 +1,174 @@
+| Code Portability | Low, tied to MetaMask | High, works across multiple wallet implementations |
+
+
+### Use `wallet_invokeSnap` directly
+
+When using `wallet_invokeSnap` directly to interact with the Starknet Snap, you typically follow the following steps:
+
+1. Ensure the Starknet Snap is installed.
+2. Construct the appropriate parameters for the Starknet operation you want to perform.
+3. Call `wallet_invokeSnap` with the specified parameters.
+
+Example for sending a transaction:
+
+```javascript title="title="starknetSnapInteraction.js"
+const SnapId = 'npm:@starknet/Snap';
+
+async function interactWithStarknetSnap() {
+  try {
+    // Ensure the Snap is installed and connected
+    await window.ethereum.request({
+      method: 'wallet_enable',
+      params: [{
+        wallet_Snap: { [SnapId]: {} },
+      }]
+    });
+
+    // Use wallet_invokeSnap to call a Starknet method
+    const result = await window.ethereum.request({
+      method: 'wallet_invokeSnap',
+      params: [
+        SnapId,
+        {
+          method: 'starknet_signAndSendTransaction',
+          params: {
+            calls: [{
+              contractAddress: '0x123...',
+              entrypoint: 'transfer',
+              calldata: ['0x456...', '1000']
+            }]
+          }
+        }
+      ]
+    });
+
+    console.log('Transaction hash:', result);
+  } catch (error) {
+    console.error('Error:', error);
+  }
+}
+```
+
+With `wallet_invokeSnap`, you must know the exact method names and parameter structures for interacting with the Starknet Snap. Additionally, you must handle both MetaMask-specific and Starknet-specific errors. Since `wallet_invokeSnap` is designed specifically for MetaMask, it operates within the framework of MetaMask interactions.
+
+### Use the `get-starknet` integration
+
+With `get-starknet`, the process is simplified:
+
+1. Connect to a wallet using `get-starknet`.
+2. Use the provided high-level methods to perform Starknet operations.
+
+Example for sending the same transaction:
+
+```javascript "title="starknetSnapInteraction.js""
+import { connect } from 'get-starknet';
+
+async function sendTransaction() {
+  try {
+    const starknet = await connect();
+    if (!starknet) throw new Error('Failed to connect to wallet');
+
+    const result = await starknet.account.execute({
+      contractAddress: '0x123...',
+      entrypoint: 'transfer',
+      calldata: ['0x456...', '1000']
+    });
+    console.log('Transaction hash:', result.transaction_hash);
+  } catch (error) {
+    console.error('Error:', error);
+  }
+}
+```
+
+`get-starknet` provides a high-level, intuitive API that abstracts complex operations and manages wallet connections. 
+It offers multi-wallet support, standardized error handling, and more readable code compared to direct Snap invocation so you are not required to manage lower-level Starknet interactions.
```

Comment:
This page should be a short, high-level intro to Starknet. These instructions should be moved to the existing "Connect to Starknet" guide – you can add the `wallet_invokeSnap` instructions underneath the `get-starknet` section.

Reply:
I was trying to highlight/compare the actions after the app is developed. I don''t think this belongs in the connect section. Maybe in manage, but the point was to show the benefits using `get-starknet. Maybe I'll just remove this completely